opendatahub-io/data-science-pipelines-operator
# feat: always enable server config and make it customizable

## Description of your changes:
We should always mount the server config, regadless of whether we have a custom cabundle or not, so that we don't end up in a situation where based on the cabundle we are using different configs, as that may result in different behaviors if somehow the default config in the dsp server image ends up diverging from the one we mount.

To prevent future issues with server config, this pr also adds the ability for users to provide a custom sever config, in case the default options are not sufficient, the user can customize their own server config in such a case, for instance changing the path in s3 where pipelines are stored.

The name for the default server config is also renamed to be consistent with the rest of the dspa's resource naming pattern "ds-pipelines-*"

## Testing instructions
1. Deploy simple dspa
2. confirm server-config is mounted, in the pipeline server pod you should see the following: 

```yaml
         volumeMounts:
            - name: server-config
              mountPath: /config/config.json
              subPath: config.json
```
3. confirm pipeline runs successfully
4. In a separate namespace, deploy another dspa but with a custom server config, for example: 
```yaml
kind: ConfigMap
apiVersion: v1
metadata:
  name: custom-script
  namespace: dspa1
  labels:
    app: ds-pipeline-test
    component: data-science-pipelines
data:
  blah: |
    {
      "DBConfig": {
        "MySQLConfig": {
          "ExtraParams": {"tls":"false"},
          "GroupConcatMaxLen": "4194304"
         },
        "PostgreSQLConfig": {},
        "ConMaxLifeTime": "120s"
      },
      "DBDriverName": "mysql",
      "InitConnectionTimeout": "6m",
      "CacheEnabled": "false"
    }

```

and in dspa: 

```yaml
apiVersion: datasciencepipelinesapplications.opendatahub.io/v1alpha1
kind: DataSciencePipelinesApplication
metadata:
  name: test
spec:
  apiServer:
    customServerConfigMap:
      name: custom-script
      key: blah
  dspVersion: v2
  objectStorage:

    minio:
      deploy: true
      image: 'quay.io/opendatahub/minio:RELEASE.2019-08-14T20-37-41Z-license-compliance'
  mlpipelineUI:
    image: quay.io/opendatahub/ds-pipelines-frontend:latest
```

then inspect the apiserver mounts: 

```yaml
 volumeMounts:
    - name: server-config
      mountPath: /config/config.json
      subPath: blah
```





## Checklist
